airflow/contrib/hooks/gcs_hook.py

```diff
@@ -177,7 +177,9 @@ def download(self, bucket, object, filename=None):
 
     # pylint:disable=redefined-builtin
     def upload(self, bucket, object, filename,
-               mime_type='application/octet-stream', gzip=False):
+               mime_type='application/octet-stream', gzip=False,
+               multipart=False, chunksize=256 * 1024 * 1024,
```

Comment:
API Design question. It would also be possible to set chunksize to None by default, it set then it will switch to multipart. This will reduce the number of options (because `multipart` is set implicitly). What do you think?

Reply:
Avoiding two options seems beneficial, but the onus is then on every call site to specify not just that the upload should be multipart, but explicitly select a size.

Would this work

```
upload(..., multipart=False)  # default, single-part upload
upload(..., multipart=True)  # multipart with default chunksize
upload(..., multipart=1024 * 1024)  # multipart with chosen chunksize
```

with `false` being the default value in the signature?